salt/modules/dnsmasq.py

```diff
@@ -165,22 +193,20 @@ def _parse_dnamasq(filename):
 
     with salt.utils.files.fopen(filename, "r") as fp_:
         for line in fp_:
-            line = salt.utils.stringutils.to_unicode(line)
-            if not line.strip():
+            line = salt.utils.stringutils.to_unicode(line).strip()
+            if not line:  # Empty string
                 continue
-            if line.startswith("#"):
+            if line.strip().startswith("#"):
```

Comment:
This can be just lstrip since you don't need to strip the end of the string to see what it starts with.

Reply:
Done in https://github.com/saltstack/salt/pull/65215/commits/5b789bde7dfc667d069eba1f3459d7bd762d60ee